modules/dop/endpoints/pipeline.go

```diff
@@ -353,7 +354,23 @@ func (e *Endpoints) pipelineRun(ctx context.Context, r *http.Request, vars map[s
 		IdentityInfo:      identityInfo,
 		PipelineRunParams: runRequest.PipelineRunParams,
 	}); err != nil {
-		return errorresp.ErrResp(err)
+		var apiError, ok = err.(*errorresp.APIError)
+		if !ok {
+			return errorresp.ErrResp(err)
+		}
+
+		ctxMap, ok := apiError.Ctx().(map[string]interface{})
+		if !ok {
+			return errorresp.ErrResp(err)
```

Comment:
please `new` a new error.

can you image where this `err` from or related to ?!